libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/ui/SalesforceReactActivity.java

```diff
@@ -276,6 +289,54 @@ protected boolean shouldReactBeRunning(){
 
     protected void restartReactNativeApp(){
         SalesforceReactActivity.this.getReactNativeHost().getReactInstanceManager().destroy();
-        SalesforceReactActivity.this.getReactNativeHost().getReactInstanceManager().createReactContextInBackground();
+        if(shouldReactBeRunning()){
+            SalesforceReactActivity.this.getReactNativeHost().getReactInstanceManager().createReactContextInBackground();
+        }
+    }
+
+    private boolean shouldAskOverlayPermission(){
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            if(SalesforceReactActivity.this.getReactNativeHost().getReactInstanceManager().getDevSupportManager().getDevSupportEnabled()){
+                if (!Settings.canDrawOverlays(this)) {
+                    showPermissionWarning();
+                    return true;
+                }
+                else{
+                    hidePermissionWarning();
+                }
+            }
+        }
+        return false;
+    }
+
+    private void loadReactAppOnceIfReady() {
+        if(reactActivityDelegate != null ){
+            reactActivityDelegate.loadReactAppOnceIfReady(getMainComponentName());
+        }
+    }
+
+    private void showPermissionWarning(){
+        if(overlayPermissionRequiredDialog == null){
```

Comment:
The actual overlay dialog is triggered here:
https://github.com/facebook/react-native/blob/0.35-stable/ReactAndroid/src/main/java/com/facebook/react/ReactActivityDelegate.java#L83

Yes. If overlay perm dialog is closed without confirming overlay draw perm: this information dialog is displayed: (I didn't want to duplicate the existing code for this corner case) 
When a user presses 'Continue' button on this informational dialogue: the activity gets recreated and  ReactActivityDelegate opens Overlay Perm screen again.